core/src/main/scala/cats/data/NonEmptyChain.scala

```diff
@@ -372,6 +372,24 @@ class NonEmptyChainOps[A](val value: NonEmptyChain[A]) extends AnyVal {
   /** Reverses this `NonEmptyChain` */
   final def reverse: NonEmptyChain[A] =
     create(toChain.reverse)
+
+  /**
+   * Remove duplicates. Duplicates are checked using `Order[_]` instance.
+   */
+  final def distinct[AA >: A](implicit O: Order[AA]): NonEmptyChain[AA] = {
```

Comment:
I wonder. Dont you think that it would be better to to call `toChain` and then `distinct` on chain and then rebuild NonEmptyChain? It would remove duplication but I'm not sure if it will still be ok with performance.

Reply:
The other way around would be cheaper, something along the lines of `NonEmptyChain.fromChain(this).fold(empty[A])(_.distinct.toChain)`